Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -286,6 +290,26 @@ private extension UInt8 {
     }
 }
 
+/* private but tests */ internal extension HTTPHeaders {
+    func isKeepAlive(version: HTTPVersion) -> Bool {
+        switch self._storage.keepAliveState {
+        case .close:
+            return false
+        case .keepAlive:
+            return true
+        case .unknown:
+            guard let connection = self["connection"].first?.lowercased() else {
```

Comment:
whilst you're working on this: worth fixing this implementation of follow-up?

Reply:
Let me do a follow up